aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,221 @@
+    String glueCatalogId = properties.get(AwsProperties.GLUE_CATALOG_ID);
+    String databaseName = properties.get(AwsProperties.LAKE_FORMATION_DB_NAME);
+    String tableName = properties.get(AwsProperties.LAKE_FORMATION_TABLE_NAME);
+
+    GetTableResponse response = glue.getTable(GetTableRequest.builder()
+        .catalogId(glueCatalogId)
+        .databaseName(databaseName)
+        .name(tableName)
+        .build());
+    boolean isRegisteredWithLakeFormation = response.table().isRegisteredWithLakeFormation() != null ?
+        response.table().isRegisteredWithLakeFormation() : Boolean.FALSE;
+
+    if (isRegisteredWithLakeFormation) {
+      return new LakeFormationCredentialsProvider(properties);
+    } else {
+      return getAwsCredentialsProvider();
+    }
+  }
+
+  private static Region region(Map<String, String> properties) {
+    String region = properties.containsKey(AwsProperties.LAKE_FORMATION_AWS_REGION) ?
+        properties.get(AwsProperties.LAKE_FORMATION_AWS_REGION) : null;
+    if (region == null || region.isEmpty()) {
+      throw new RuntimeException("Region is empty");
+    }
+    return Region.of(region);
+  }
+
+  private AwsCredentialsProvider getAwsCredentialsProvider() {
+    return StaticCredentialsProvider.create(AwsSessionCredentials.create(accessKeyId, secretAccessKey, sessionToken));
+  }
+
+  static class LakeFormationCredentialsProvider implements AwsCredentialsProvider {
+    private LakeFormationClient lakeFormationClient;
+
+    private String tableArn;
+    private static final List<String> privilegeList = Arrays.asList("SELECT", "INSERT", "DELETE");
+
+    LakeFormationCredentialsProvider(Map<String, String> properties) {
+      AwsSessionCredentials awsSessionCredentials = getAssumeRoleCredentials(properties);
+      String region = properties.containsKey(AwsProperties.LAKE_FORMATION_AWS_REGION) ?
+          properties.get(AwsProperties.LAKE_FORMATION_AWS_REGION) : null;
+
+      lakeFormationClient = buildLakeFormationClient(awsSessionCredentials.accessKeyId(),
+          awsSessionCredentials.secretAccessKey(), awsSessionCredentials.sessionToken(), region);
+      tableArn = getTableArn(properties);
+    }
+
+    @Override
+    public AwsCredentials resolveCredentials() {
+      GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+          GetTemporaryGlueTableCredentialsRequest.builder()
+              .tableArn(tableArn)
+              .permissionsWithStrings(privilegeList)
+              .build();
+      GetTemporaryGlueTableCredentialsResponse response =
+          lakeFormationClient.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+    }
+
+    private AwsSessionCredentials getAssumeRoleCredentials(Map<String, String> properties) {
+      String roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
```

Comment:
I think we should not use the assume role properties for this purpose. That is used by ALL clients in the assume role client factory, but in this context we are only using it for LakeFormation client. I think the following set of property names make more sense:
- lakeformation.role.arn
- lakeformation.role.external-id
- lakeformation.role.region
- lakeformation.role.timeout-sec

Reply:
discussed offline, using different configs for LF only is clearer, but reusing the same config saves 4 config keys, and the tags feature can be reused. In this client factory, only LF is using this assume role config key values, so it does not have confusion. This sounds like a better solution, so we can keep the config keys as is instead of having new ones for LF client role only.